enumerables.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array
+  end
+
+  def my_all?
+    selfitem = self
+    selfitem.my_each { |x| return false unless yield(x) }
+    true
+  end
+
+  def my_any?
+    selfitem = self
+    selfitem.my_each { |x| return true if yield(x) }
+    false
+  end
```

Comment:
enumerables my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
     Failure/Error: expect(true_array.my_any?).to be true_array.any?

enumerables my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
     Failure/Error: expect(array.my_any?(Integer)).to be array.any?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
     
     LocalJumpError:
       no block given (yield)

enumerables my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
      Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

 enumerables my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)